python/cuml/experimental/common/base.pyx

```diff
@@ -0,0 +1,548 @@
+                    # if import path differs from the one of sklearn
+                    # look for sk_import_path_
+                    model_path = self.sk_import_path_
+                else:
+                    # import from similar path to the current estimator
+                    # class
+                    model_path = 'sklearn' + self.__class__.__module__[4:]
+                model_name = self.__class__.__name__
+                sk_model = getattr(import_module(model_path), model_name)
+                # initialize model
+                self.sk_model_ = sk_model()
+                # transfer params set during cuml estimator initialization
+                for param in self.get_param_names():
+                    self.sk_model_.__dict__[param] = self.__dict__[param]
+
+                # transfer attributes trained with cuml
+                for attribute in self.get_attributes_names():
+                    # check presence of attribute
+                    if hasattr(self, attribute):
+                        # get the cuml attribute
+                        cu_attr = self.__dict__[attribute]
+                        # if the cuml attribute is a CumlArrayDescriptorMeta
+                        if hasattr(cu_attr, 'get_input_value'):
+                            # extract the actual value from the
+                            # CumlArrayDescriptorMeta
+                            cu_attr_value = cu_attr.get_input_value()
+                            # check if descriptor is empty
+                            if cu_attr_value is not None:
+                                if cu_attr.input_type == 'cuml':
+                                    # transform cumlArray to numpy and set it
+                                    # as an attribute in the sklearn model
+                                    self.sk_model_.__dict__[attribute] = \
+                                        cu_attr_value.to_output('numpy')
+                                else:
+                                    # transfer all other types of attributes
+                                    # directly
+                                    self.sk_model_.__dict__[attribute] = \
+                                        cu_attr_value
+                        else:
+                            # transfer all other types of attributes directly
+                            self.sk_model_.__dict__[attribute] = cu_attr
+                    else:
+                        raise ValueError('Attribute could not get transfered')
+
+            # helper function to convert non-builtin as numpy arrays
+            def to_host(data):
+                if isinstance(data, (int, float, complex, bool, str,
+                              type(None), dict, set, list, tuple)):
+                    return data
+                else:
+                    try:
+                        return input_to_host_array(data)[0]
+                    except Exception as e:
+                        return data
```

Comment:
Seems like this functionality could be easily included in `input_to_host_array`

Also not sure if this path 

```python
except Exception as e:
                         return data
```

is actually safe